pkg/engine/loader.go

```diff
@@ -74,3 +86,75 @@ func (e *Engine) LoadConstraintsFromFile(path string) (map[constraints.Constrain
 	}
 	return constraints.ParseConstraintsFromFile(bytesArr)
 }
+
+func correctPointers(source core.BaseConstruct, resourceMap map[core.ResourceId]core.BaseConstruct) error {
+	sourceValue := reflect.ValueOf(source)
+	sourceType := sourceValue.Type()
+	if sourceType.Kind() == reflect.Pointer {
+		sourceValue = sourceValue.Elem()
+		sourceType = sourceType.Elem()
+	}
+	for i := 0; i < sourceType.NumField(); i++ {
+		fieldValue := sourceValue.Field(i)
+		switch fieldValue.Kind() {
+		case reflect.Slice, reflect.Array:
+			for elemIdx := 0; elemIdx < fieldValue.Len(); elemIdx++ {
+				elemValue := fieldValue.Index(elemIdx)
+
+				loadNestedResourcesFromIds(source, elemValue, resourceMap)
+			}
+
+		case reflect.Map:
+			for iter := fieldValue.MapRange(); iter.Next(); {
+				elemValue := iter.Value()
+				loadNestedResourcesFromIds(source, elemValue, resourceMap)
+			}
+
+		default:
+			loadNestedResourcesFromIds(source, fieldValue, resourceMap)
+		}
+	}
+	return nil
+}
+
+func loadNestedResourcesFromIds(source core.BaseConstruct, targetValue reflect.Value, resourceMap map[core.ResourceId]core.BaseConstruct) {
```

Comment:
Does this one differ from the one in `core`?

Reply:
yeah changed the name to explain the difference

Reply:
and added some docs